src/clj/puppetlabs/trapperkeeper/services/watcher/filesystem_watch_core.clj

```diff
@@ -30,9 +30,11 @@
   [event :- WatchEvent
    watch-path :- Path]
   {:type (get event-type-mappings (.kind event))
-   :path (-> watch-path
-             (.resolve (.context event))
-             fs/file)})
+   :path (if (= StandardWatchEventKinds/OVERFLOW (.kind event))
+           (fs/file (.resolve watch-path "UNKNOWN"))
```

Comment:
I think this is wrong because it is trying to resolve a file named `UNKNOWN`.  Or, I am not understanding something?

I was thinking we'd need to deal with this by making `:path` either a `schema/optional-key` or `(schema/maybe File)` in the `Event` schema.  Or, something more drastic:
```clj
(def Event
  "Schema for an event on a file watched by this service."
  (schema/if #(= (:type %) :unknown)
    {:type (schema/eq :unknown)
     :path (schema/maybe File)}
    {:type (schema/enum :create :modify :delete)
     :path File}))
```